src/Client/Client.fs

```diff
@@ -0,0 +1,63 @@
+namespace TcpEcho
+
+open System
+open System.Buffers
+open System.Net.Sockets
+open System.IO.Pipelines
+open System.Text
+open System.Threading.Tasks
+
+[<AbstractClass>]
+type Client(endpoint: string, port: int) =
+    let minimumBufferSize = 1024
+
+    let mkString (buffer: ReadOnlySequence<byte>) =
+        seq { for segment in buffer -> segment.ToArray() |> Encoding.UTF8.GetString } |> Seq.fold (+) ""
+
+    let rec writeToPipeAsync (writer: PipeWriter) (socket: Socket) = async {
+        let segment = Array.zeroCreate<byte> minimumBufferSize |> ArraySegment
+        let! read = socket.ReceiveAsync(segment, SocketFlags.None) |> Async.AwaitTask
+        match read with
+        | 0 -> return writer.Complete()
+        | bytesRead ->
+            segment.Array.CopyTo(writer.GetMemory(bytesRead))
+            writer.Advance bytesRead
+            let! flusher = writer.FlushAsync().AsTask() |> Async.AwaitTask
+            if flusher.IsCompleted then return writer.Complete()
+
+        match socket.Available with
+        | 0 -> return writer.Complete()
+        | _ -> return! writeToPipeAsync writer socket
+    }
+
+    let rec readFromPipeAsync (reader: PipeReader) str = async {
+        let! result = reader.ReadAsync().AsTask() |> Async.AwaitTask
+        let buffer : ReadOnlySequence<byte> = result.Buffer
```

Comment:
nit(cosmetic): you used no spaces before colons in other places, so let's be consistent in this line above too